### ISSUE TITLE
'SPXException', reason: 'Exception with an error code: 0x15 (SPXERR_MIC_ERROR)

### ISSUE THREAD

**Issue by fsaar** [Azure-Samples/cognitive-services-speech-sdk#514]
Encountered an exception in the native iOS SDK V1.9.0
Integrated the SDK via a custom cordova plugin into an Ionic / Angular project. BTW had to fall back to the ios native SDK version since the javascript framework is not usable on iOS due to lacking support for 'getUserMedia()' in UIWebview / wkwebview. Works only in Safari and SFSafariViewController.

So,when I access the the microphone on iOS (first time the mic access dialog correctly pops up) I see the exception.
 Cordova Platforms  : android 7.1.4 ios 4.5.5
    Ionic Framework    : ionic-angular 3.9.2
 Xcode      : Xcode 11.3.1 Build version 11C505 

- OS: iOS
 - Hardware -  ARM
 - Programming language: Typescript / Javascript / Angular / Ionic / Cordova / Objective-C

The native side of the plugin is written in Objective-C
The cordova plugin interface in Javascript 
The Ionic / Angular implementation is based on typescript.

Here's the callstack that's been dumped into the console along with the exception
2020-02-20 10:44:49.784722+0000 XXXXXXXX[2875:2886562] Exception caught in core: Exception with an error code: 0x15 (SPXERR_MIC_ERROR)
[CALL STACK BEGIN]

3   MicrosoftCognitiveServicesSpeech    0x000000010245d5f8 file_logger_stop_logging + 795588
4   MicrosoftCognitiveServicesSpeech    0x0000000102461778 file_logger_stop_logging + 812356
5   MicrosoftCognitiveServicesSpeech    0x00000001023a8838 file_logger_stop_logging + 54788
6   MicrosoftCognitiveServicesSpeech    0x00000001023a6ddc file_logger_stop_logging + 48040
7   MicrosoftCognitiveServicesSpeech    0x00000001023c3360 file_logger_stop_logging + 164140
8   MicrosoftCognitiveServicesSpeech    0x000000010241b1e4 file_logger_stop_logging + 524208
9   MicrosoftCognitiveServicesSpeech    0x000000010241b170 file_logger_stop_logging + 524092
10  MicrosoftCognitiveServicesSpeech    0x000000010241cc7c file_logger_stop_logging + 531016
11  MicrosoftCognitiveServicesSpeech    0x000000010241b350 file_logger_stop_logging + 524572
12  MicrosoftCognitiveServicesSpeech    0x000000010241e320 file_logger_stop_logging + 536812
13  libsystem_pthread.dylib             0x00000001988de1d0 _pthread_start + 124
14  libsystem_pthread.dylib             0x00000001988e1ae0 thread_start + 8
[CALL STACK END]
2020-02-20 10:44:49.786592+0000 XXXXXX [2875:2886562] *** Terminating app due to uncaught exception 'SPXException', reason: 'Exception with an error code: 0x15 (SPXERR_MIC_ERROR)
[CALL STACK BEGIN]

Here's the callstack via lldb's thread backtrace:

thread #1, queue = 'com.apple.main-thread', stop reason = signal SIGABRT
    frame #0: 0x00000001989b9ec4 libsystem_kernel.dylib`__pthread_kill + 8
    frame #1: 0x00000001988d5774 libsystem_pthread.dylib`pthread_kill$VARIANT$mp + 112
    frame #2: 0x0000000198829844 libsystem_c.dylib`abort + 100
    frame #3: 0x00000001989827d4 libc++abi.dylib`abort_message + 128
    frame #4: 0x00000001989829c4 libc++abi.dylib`demangling_terminate_handler() + 296
    frame #5: 0x00000001988ea258 libobjc.A.dylib`_objc_terminate() + 124
    frame #6: 0x000000019898f304 libc++abi.dylib`std::__terminate(void (*)()) + 16
    frame #7: 0x000000019898ec58 libc++abi.dylib`__cxxabiv1::failed_throw(__cxxabiv1::__cxa_exception*) + 32
    frame #8: 0x000000019898ec18 libc++abi.dylib`__cxa_throw + 124
    frame #9: 0x00000001988ea0d0 libobjc.A.dylib`objc_exception_throw + 356
    frame #10: 0x0000000198ac5eb0 CoreFoundation`-[NSException raise] + 12
    frame #11: 0x00000001022ac988 MicrosoftCognitiveServicesSpeech`___lldb_unnamed_symbol874$$MicrosoftCognitiveServicesSpeech + 1200
  * frame #12: 0x00000001007e5ca4 XXXXXXXX`-[CDVSpeechRec recognizeFromMicContinuous](self=0x000000028365cf00, _cmd="recognizeFromMicContinuous") at CDVSpeechRec.m:128:9
    frame #13: 0x00000001007e58f4 XXXXXX`-[CDVSpeechRec startTranscriptionContinuous:](self=0x000000028365cf00, _cmd="startTranscriptionContinuous:", command=0x0000000281a13f00) at CDVSpeechRec.m:89:5
    frame #14: 0x000000010092ee94 XXXXXX `-[CDVCommandQueue execute:](self=0x0000000281bd13b0, _cmd="execute:", command=0x0000000281a13f00) at CDVCommandQueue.m:181:9
    frame #15: 0x00000001007689f8 XXXXXX`-[CDVWKWebViewEngine handleCordovaMessage:](self=0x0000000283b24200, _cmd="handleCordovaMessage:", message=0x0000000281a10f60) at CDVWKWebViewEngine.m:613:10
    frame #16: 0x0000000100768860 XXXXXX`-[CDVWKWebViewEngine userContentController:didReceiveScriptMessage:](self=0x0000000283b24200, _cmd="userContentController:didReceiveScriptMessage:", userContentController=0x0000000282231b80, message=0x0000000281a10f60) at CDVWKWebViewEngine.m:599:9
    frame #17: 0x000000010076a800 XXXXXX`-[CDVWKWeakScriptMessageHandler userContentController:didReceiveScriptMessage:](self=0x00000002817229b0, _cmd="userContentController:didReceiveScriptMessage:", userContentController=0x0000000282231b80, message=0x0000000281a10f60) at CDVWKWebViewEngine.m:807:5
    frame #18: 0x00000001a02e4630 WebKit`ScriptMessageHandlerDelegate::didPostMessage(WebKit::WebPageProxy&, WebKit::FrameInfoData const&, WebCore::SerializedScriptValue&) + 228
    frame #19: 0x00000001a046a79c WebKit`WebKit::WebUserContentControllerProxy::didPostMessage(IPC::Connection&, WTF::ObjectIdentifier<WebCore::PageIdentifierType>, WebKit::FrameInfoData const&, unsigned long long, IPC::DataReference const&) + 172
    frame #20: 0x00000001a063b710 WebKit`WebKit::WebUserContentControllerProxy::didReceiveMessage(IPC::Connection&, IPC::Decoder&) + 132
    frame #21: 0x00000001a0162ffc WebKit`IPC::MessageReceiverMap::dispatchMessage(IPC::Connection&, IPC::Decoder&) + 104
    frame #22: 0x00000001a03b24c0 WebKit`WebKit::WebProcessProxy::didReceiveMessage(IPC::Connection&, IPC::Decoder&) + 32
    frame #23: 0x00000001a014d644 WebKit`IPC::Connection::dispatchMessage(std::__1::unique_ptr<IPC::Decoder, std::__1::default_delete<IPC::Decoder> >) + 204
    frame #24: 0x00000001a01504a0 WebKit`IPC::Connection::dispatchIncomingMessages() + 404
    frame #25: 0x00000001a789e55c JavaScriptCore`WTF::RunLoop::performWork() + 276
    frame #26: 0x00000001a789e81c JavaScriptCore`WTF::RunLoop::performWork(void*) + 36
    frame #27: 0x0000000198b3f7c4 CoreFoundation`__CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__ + 24
    frame #28: 0x0000000198b3f71c CoreFoundation`__CFRunLoopDoSource0 + 80
    frame #29: 0x0000000198b3eeb4 CoreFoundation`__CFRunLoopDoSources0 + 180
    frame #30: 0x0000000198b3a000 CoreFoundation`__CFRunLoopRun + 1080
    frame #31: 0x0000000198b398a0 CoreFoundation`CFRunLoopRunSpecific + 464
    frame #32: 0x00000001a2a91328 GraphicsServices`GSEventRunModal + 104
    frame #33: 0x000000019cc2a740 UIKitCore`UIApplicationMain + 1936
    frame #34: 0x00000001007336e0 XXXXXX`main(argc=1, argv=0x000000016f7078c8) at main.m:32:22
    frame #35: 0x00000001989c4360 libdyld.dylib`start + 4

Here's the relevant objective-c code to initialise the recognizer:
- (SPXSpeechRecognizer*) recognizer {
 NSString *sub ...
NSString *region = @"uksouth";
 SPXSpeechConfiguration *speechConfig = [[SPXSpeechConfiguration alloc] initWithSubscription:sub region:region];
    speechConfig.speechRecognitionLanguage = @"en-US";
    SPXAudioConfiguration *audioConfig = [[SPXAudioConfiguration alloc] init];
    SPXSpeechRecognizer *reco = [[SPXSpeechRecognizer alloc] initWithSpeechConfiguration:speechConfig audioConfiguration:audioConfig];
}

 __weak __typeof(self) weakSelf = self;
    SPXSpeechRecognizer *rec = [self recognizer];
    if (rec!= nil) {
        self.transcribing = YES;
        self.currentContinuousRecognizer = rec;
        [rec addRecognizingEventHandler:^(SPXSpeechRecognizer * recognizer, SPXSpeechRecognitionEventArgs * eventArgs) {
            NSString *recognized = eventArgs.result.text;
            weakSelf.message = recognized;
        }];
        [rec addRecognizedEventHandler:^(SPXSpeechRecognizer * recognizer, SPXSpeechRecognitionEventArgs * eventArgs) {
            NSString *recognized = eventArgs.result.text;
            weakSelf.message = recognized;
        }];
        [rec addCanceledEventHandler:^(SPXSpeechRecognizer * recognizer, SPXSpeechRecognitionCanceledEventArgs * eventArgs) {
            weakSelf.transcribing = NO;
            weakSelf.currentContinuousRecognizer = nil;
        }];
        
        [self.currentContinuousRecognizer startContinuousRecognition];
    }

The exception is thrown when calling  'startContinuousRecognition'
So something went wrong in the Audio capture system. How can I work around this?Any ideas?

**Comment by oscholz** [comment]
@jhakulin could you please have a look at this?

**Comment by oscholz** [comment]
@fsaar thank you for reporting this. Sorry for the prolonged radio silence. I have added this bug to our backlog and we will let you know what we find. We will update this issue once the associated bug is closed.

**Comment by oscholz** [comment]
@fsaar did you enter into the manifest that you need to use the microphone? 
Please let us know!

**Comment by shashankkulshrestha-kiwi** [comment]
@oscholz Did you resolve the issue? I am facing this right after the app launch. Please update

**Comment by shashankkulshrestha-kiwi** [comment]
@fsaar @oscholz  I am stuck in the issue, my client is not able to release the app. We've provided all the required microphone permission. This issue appears only after 1st installation and works fine after that. Can you please let us know the possible issue or solution?

**Comment by jhakulin** [comment]
@shashankkulshrestha-kiwi Could you please report a new issue, including details on what is the programming language and platform where your app is running ? Thank you.
